dataframe-arrow/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/ArrowKtTest.kt

```diff
@@ -0,0 +1,237 @@
+
+        assertEstimations(DataFrame.readArrowFeather(testArrowFeather("test-with-nulls.arrow"), NullabilityOptions.Widening), true, true)
+        assertEstimations(DataFrame.readArrowIPC(testArrowIPC("test-with-nulls.arrow"), NullabilityOptions.Widening), true, true)
+    }
+
+    @Test
+    fun testReadingAllTypesAsEstimatedNotNullable() {
+        assertEstimations(DataFrame.readArrowFeather(testArrowFeather("test-not-nullable.arrow"), NullabilityOptions.Infer), false, false)
+        assertEstimations(DataFrame.readArrowIPC(testArrowIPC("test-not-nullable.arrow"), NullabilityOptions.Infer), false, false)
+
+        assertEstimations(DataFrame.readArrowFeather(testArrowFeather("test-not-nullable.arrow"), NullabilityOptions.Checking), false, false)
+        assertEstimations(DataFrame.readArrowIPC(testArrowIPC("test-not-nullable.arrow"), NullabilityOptions.Checking), false, false)
+
+        assertEstimations(DataFrame.readArrowFeather(testArrowFeather("test-not-nullable.arrow"), NullabilityOptions.Widening), false, false)
+        assertEstimations(DataFrame.readArrowIPC(testArrowIPC("test-not-nullable.arrow"), NullabilityOptions.Widening), false, false)
+    }
+
+    @Test
+    fun testReadingAllTypesAsEstimatedNotNullableWithNulls() {
+        assertEstimations(DataFrame.readArrowFeather(testArrowFeather("test-illegal.arrow"), NullabilityOptions.Infer), true, true)
+        assertEstimations(DataFrame.readArrowIPC(testArrowIPC("test-illegal.arrow"), NullabilityOptions.Infer), true, true)
+
+        shouldThrow<IllegalArgumentException> {
+            assertEstimations(DataFrame.readArrowFeather(testArrowFeather("test-illegal.arrow"), NullabilityOptions.Checking), false, true)
+        }
+        shouldThrow<IllegalArgumentException> {
+            assertEstimations(DataFrame.readArrowIPC(testArrowIPC("test-illegal.arrow"), NullabilityOptions.Checking), false, true)
+        }
+
+        assertEstimations(DataFrame.readArrowFeather(testArrowFeather("test-illegal.arrow"), NullabilityOptions.Widening), true, true)
+        assertEstimations(DataFrame.readArrowIPC(testArrowIPC("test-illegal.arrow"), NullabilityOptions.Widening), true, true)
+    }
+
+    @Test
+    fun testWritingGeneral() {
+        fun assertEstimation(citiesDeserialized: DataFrame<*>) {
+            citiesDeserialized["name"] shouldBe citiesExampleFrame["name"]
+            citiesDeserialized["affiliation"] shouldBe citiesExampleFrame["affiliation"]
+            citiesDeserialized["is_capital"] shouldBe citiesExampleFrame["is_capital"]
+            citiesDeserialized["population"] shouldBe citiesExampleFrame["population"]
+            citiesDeserialized["area"] shouldBe citiesExampleFrame["area"]
+            citiesDeserialized["settled"].type() shouldBe typeOf<LocalDate>() // cities["settled"].type() refers to FlexibleTypeImpl(LocalDate..LocalDate?) and does not match typeOf<LocalDate>()
```

Comment:
Nice catch, i think it should be fixed in `readArrow` later as well

Reply:
`FlexibleTypeImpl` comes not from Arrow but from `listOf(LocalDate.of())` in the sample generation.
What exactly do you want to fix?